packages/sit-onyx/src/components/OnyxDataGrid/features/filtering/types.ts

```diff
@@ -0,0 +1,70 @@
+import type { MaybeRefOrGetter } from "vue";
+import type { DataGridEntry } from "../../types";
+
+/**
+ * Configuration for filtering on a per-column basis.
+ * If at least one column has a filtering configuration, filtering must be explicitly enabled for all columns.
+ */
+export type FilterColumnOptions<TEntry extends DataGridEntry> = {
+  [TKey in keyof TEntry]?: {
+    /**
+     * Determines if filtering is enabled for this column.
+     * If set to true, the column will be filterable.
+     */
+    enabled: boolean;
+    /**
+     * The filter value for this column.
+     * This value will be used to filter the data in this column.
+     */
+    searchTerm?: string;
+    /**
+     * Configuration for how filtering should behave for this column.
+     */
+    config?: FilterConfig;
+  };
+};
+/**
+ * Configuration for how filtering should behave.
+ */
+export type FilterConfig = {
+  /**
+   * A custom filtering function for this column.
+   * This function is used to filter the column data, instead of using the default filtering behavior.
+   */
+  filterFunc?: (entryValue: string, searchTerm: string) => boolean;
```

Comment:
Please use the filter function that is defined in #1941 :)

```ts
  filterFunc?: (
    /**
     * The searchTerm as entered by the user.
     */
    searchTerm: string,
    /**
     * The current row value to be checked.
     */
    value: TEntry[keyof TEntry],
    /**
     * The column that is filtered by.
     */
    column: keyof TEntry,
    /**
     * The complete row data.
     */
    entry: TEntry,
```

We should pass additional data